fdbserver/TLogServer.actor.cpp

```diff
@@ -2444,6 +2457,13 @@ ACTOR Future<Void> rejoinClusterController(TLogData* self,
 			if (BUGGIFY)
 				wait(delay(SERVER_KNOBS->BUGGIFY_WORKER_REMOVED_MAX_LAG * deterministicRandom()->random01()));
 			throw worker_removed();
+		} else if (inf.recoveryCount > recoveryCount && stoppedPromise.canBeSet()) {
```

Comment:
Or, is this the main change that solves the race?

Reply:
This resolves the second bug.

Reply:
This is so I understand it better: how is it better than checking a boolean flag?

Reply:
This promise is needed at the other place so that we can wait on `logData->stoppedPromise.getFuture()`, which can't be achieved with a bool.